cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,258 @@
+        //  service/idbrokermappingmanagement/server/IdBrokerMappingManagementService.java#L109
+        List<String> dataAccessUsers = Arrays.asList("hbase", "hdfs", "hive", "impala", "yarn",
+            "dpprofiler", "zeppelin", "kudu", "hue");
+
+        AccountMappingBase accountMappings = cloudFileSystem.getAccountMapping();
+        if (accountMappings != null) {
+            Set<String> roleArns = getRoleArnsForUsers(dataAccessUsers, accountMappings.getUserMappings());
+
+            List<String> policyFileNames = Arrays.asList(
+                "aws-cdp-bucket-access-policy.json",
+                "aws-cdp-datalake-admin-s3-policy.json"
+            );
+            String s3GuardDynamoDBTableName = cloudFileSystem.getS3GuardDynamoTableName();
+            if (s3GuardDynamoDBTableName != null) {
+                policyFileNames.add("aws-cdp-dynamodb-policy.json");
+            } else {
+                s3GuardDynamoDBTableName = "";
+            }
+
+            for (StorageLocationBase location : cloudFileSystem.getLocations()) {
+                String storageLocationBase = location.getValue().replace("s3a://", "");
+                String datalakeBucket = storageLocationBase.split("/", 2)[0];
+                checkRolePermissions(roleArns, policyFileNames, datalakeBucket,
+                    storageLocationBase, s3GuardDynamoDBTableName);
+            }
+        }
+    }
+
+    private void checkRangerAuditRolePermissions(CloudS3View cloudFileSystem) {
+        // This list needs to match the IDBMMS list of baseline services
+        // https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/
+        //   services/idbrokermappingmanagement/src/main/java/com/cloudera/thunderhead/
+        //   service/idbrokermappingmanagement/server/IdBrokerMappingManagementService.java#L113
+        List<String> rangerAuditUsers = Arrays.asList("kafka", "solr", "knox", "atlas");
+
+        AccountMappingBase accountMappings = cloudFileSystem.getAccountMapping();
+        if (accountMappings != null) {
+            Set<String> roleArns = getRoleArnsForUsers(rangerAuditUsers, accountMappings.getUserMappings());
+
+            List<String> policyFileNames = Arrays.asList(
+                "aws-cdp-bucket-access-policy.json",
+                "aws-cdp-ranger-audit-s3-policy.json"
+            );
+            String s3GuardDynamoDBTableName = cloudFileSystem.getS3GuardDynamoTableName();
+            if (s3GuardDynamoDBTableName != null) {
+                policyFileNames.add("aws-cdp-dynamodb-policy.json");
+            } else {
+                s3GuardDynamoDBTableName = "";
+            }
+
+            for (StorageLocationBase location : cloudFileSystem.getLocations()) {
+                if (CloudStorageCdpService.RANGER_AUDIT.equals(location.getType())) {
+                    String storageLocationBase = location.getValue()
+                                                    .replace("s3a://", "")
```

Comment:
FileSystemType class contains the protocol maybe we should use that here

Reply:
Latest pushed change removes hardcoded `s3a`